### DISCUSSION TITLE
WSL- Unable to Install Foundry with Error -curl: (28) SSL connection timeout

### DISCUSSION THREAD

**Discussion by debay1010** [Cyfrin/foundry-full-course-cu#808] [Q&A] [answered]
I am finding it difficult to install foundry on my wsl-powered windows system. The command-
```
curl -L https://foundry.paradigm.xyz | bash   
 ```  
errors out after exactly 5 minutes. 

The error message simply says :
```
curl: (28) SSL connection timeout
```
please see the screenshot:
![foundry install problem](https://github.com/Cyfrin/foundry-full-course-f23/assets/143723136/fc938e60-a4fd-46ea-a5b8-2a3e620747dd)

**Comment by theirrationalone** [comment]
Hi there!

If You're using WSL, Please Follow images given below. However, You will need to click on `change settings` to change states.

![firewall step 1](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/63d20d6a-e210-4039-95a8-59be838c8c5e)
![firewall step 2](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/72cfc0a9-1450-4716-8efa-0c64ecefde38)
![firewall step 3](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/1a97c77d-72e9-46ae-ae3d-144ed5ae1aa8)
![firewall step 4](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/bebe832a-be8e-42f8-8b69-eeedcd2b50fd)
![firewall step 5](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/540caffa-07d8-40d6-af18-a8d13ea04055)
![firewall step 6](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/c4875d1d-dcc3-4b81-89a0-34903d86f1ef)
![firewall step 7](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/ebf6dd16-449b-4cb8-8ef7-3cee279ec9c7)

**Reply by debay1010** [reply]
@theirrationalone, thanks for your reply. 
However, when I checked the screenshots you gave above, they are exactly as what is already configured by default on my Firewall & Network Protection, except for the node,js part that I could not find on my system . 

And I am still having issue with the command 
```
curl -L https://foundry.paradigm.xyz | bash
```
Any other thing I can try.?

Thanks for you offer of help.

**Comment by debay1010** [comment, marked as answer]
I think I have found solution to this problem. It is simply because ssh was not install by default on the WSL. After I installed ssh, I was able to run the command.

So for anybody having this type of problem, first check if ssh is installed by executing this command 
```
sudo service ssh status
```
you should have something like below: 
```
Sep 08 11:44:33 DESKTOP-TEGL7PP systemd[1]: Starting OpenBSD Secure Shell server...
Sep 08 11:44:33 DESKTOP-TEGL7PP sshd[887]: Server listening on 0.0.0.0 port 22.
Sep 08 11:44:33 DESKTOP-TEGL7PP sshd[887]: Server listening on :: port 22.
Sep 08 11:44:33 DESKTOP-TEGL7PP systemd[1]: Started OpenBSD Secure Shell server.
```

if not then execute this command to install ssh 
```
sudo apt-get install openssh-server
```

Then you can run
```
curl -L https://foundry.paradigm.xyz | bash
```
,,, and it should go seamlessly.

**Reply by Kanishka227** [reply]
after using the sudo apt-get install openssh-server command i am getting the below status and i am still unable to install foundry

``` 
ssh.service - OpenBSD Secure Shell server
     Loaded: loaded (/lib/systemd/system/ssh.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-12-06 01:05:22 IST; 1min 53s ago
       Docs: man:sshd(8)
             man:sshd_config(5)
    Process: 193 ExecStartPre=/usr/sbin/sshd -t (code=exited, status=0/SUCCESS)

**Reply by debay1010** [reply]
That message does not look like error message to me. To confirm is ssh is now running do 
```
sudo service ssh status
```
and compare the result to what is shown above. You might be having another problem entirely different from ssh issue.

**Reply by Kanishka227** [reply]
Yes you are right i figured out the problem which was that i wasn't using a vpn . Thus i installed proton vpn and everything went smoothly. Thanks

**Reply by Moses-Aduojo** [reply]
This did not resolve the problem for me. please any other solution?

**Comment by debay1010** [comment]
Finally solved as above

**Comment by theirrationalone** [comment]
Ooh yes, that openssh i also faced that though. Great work. 🔥🔥

**Comment by AVI-SHUKLA1110** [comment]
is foundrup really slow to install ? or its just my system

**Reply by theirrationalone** [reply]
Typically, It Depends on your network speed ;`)

**Reply by Moses-Aduojo** [reply]
still having the same issue after running sudo apt-get install openssh-server
please I need help

**Comment by Moses-Aduojo** [comment]
> I am finding it difficult to install foundry on my wsl-powered windows system. The command-
> 
> ```
> curl -L https://foundry.paradigm.xyz | bash   
> ```
> 
> errors out after exactly 5 minutes.
> 
> The error message simply says :
> 
> ```
> curl: (28) SSL connection timeout
> ```
> 
> please see the screenshot: ![foundry install problem](https://private-user-images.githubusercontent.com/143723136/265940453-fc938e60-a4fd-46ea-a5b8-2a3e620747dd.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1NDYxNDMsIm5iZiI6MTcwNDU0NTg0MywicGF0aCI6Ii8xNDM3MjMxMzYvMjY1OTQwNDUzLWZjOTM4ZTYwLWE0ZmQtNDZlYS1hNWI4LTJhM2U2MjA3NDdkZC5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNlQxMjU3MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZGJkNWNlNjU5YWIxOGYzMzZiZGFkNGNkNjA4NDk3YzI5NDEwYjZhNDVlYWNkM2YyMDllZTUzY2IzZGMyMmFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BhW8YAR_OhKN7BUd0mG6E02HhlKOcOf6ckTwP6MZe6U)
